modules/cloud-run-v2/variables.tf

```diff
@@ -0,0 +1,251 @@
+  type        = string
+  default     = null
+  validation {
+    condition = (
+      var.launch_stage == null ? true : contains(
+        ["UNIMPLEMENTED", "PRELAUNCH", "EARLY_ACCESS", "ALPHA", "BETA",
+      "GA", "DEPRECATED"], var.launch_stage)
+    )
+    error_message = <<EOF
+    The launch stage should be one of UNIMPLEMENTED, PRELAUNCH, EARLY_ACCESS, ALPHA,
+    BETA, GA, DEPRECATED.
+    EOF
+  }
+}
+
+variable "name" {
+  description = "Name used for Cloud Run service."
+  type        = string
+}
+
+variable "prefix" {
+  description = "Optional prefix used for resource names."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.prefix != ""
+    error_message = "Prefix cannot be empty, please use null instead."
+  }
+}
+
+variable "project_id" {
+  description = "Project id used for all resources."
+  type        = string
+}
+
+variable "region" {
+  description = "Region used for all resources."
+  type        = string
+  default     = "europe-west1"
+}
+
+variable "revision" {
+  description = "Revision template configurations."
+  type = object({
+    name                       = optional(string)
+    gen2_execution_environment = optional(bool)
+    max_concurrency            = optional(number)
+    max_instance_count         = optional(number)
+    min_instance_count         = optional(number)
+    vpc_access = optional(object({
+      connector = optional(string)
+      egress    = optional(string)
+      subnet    = optional(string)
+      tags      = optional(list(string))
+    }))
+    timeout = optional(string)
+  })
+  default  = {}
+  nullable = false
+  validation {
+    condition = (
+      try(var.revision.vpc_access.egress, null) == null ? true : contains(
+      ["ALL_TRAFFIC", "PRIVATE_RANGES_ONLY"], var.revision.vpc_access.egress)
+    )
+    error_message = "Egress should be one of ALL_TRAFFIC, PRIVATE_RANGES_ONLY."
+  }
+}
+
+variable "service_account" {
+  description = "Service account email. Unused if service account is auto-created."
+  type        = string
+  default     = null
+}
+
+variable "service_account_create" {
+  description = "Auto-create service account."
+  type        = bool
+  default     = false
+}
+
+variable "traffic" {
+  description = "Traffic steering configuration."
+  type = map(object({
+    latest  = optional(bool)
+    percent = optional(number)
+    tag     = optional(string)
+  }))
+  default  = {}
+  nullable = false
+}
+
+variable "volumes" {
+  description = "Named volumes in containers in name => attributes format."
+  type = map(object({
+    secret = optional(object({
+      secret       = string
+      default_mode = optional(string)
+      path         = optional(string)
+      version      = optional(string)
+      mode         = optional(string)
+    }))
+    cloud_sql_instances = optional(list(string))
+    empty_dir_size      = optional(string)
+  }))
+  default  = {}
+  nullable = false
+}
+
+variable "vpc_connector_create" {
```

Comment:
Just to continue discussion about this in proper place:
1. Can we extract this variable to `variables-vpcconnector.tf` and creation of the connector also to separate `*.tf` file?
2. Then we can "upgrade" Cloud Function to give more control on the connector creation and easily sync those two by keeping these two files the same. We are experimenting with this approach in Load balancer modules ([1](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/modules/net-lb-app-ext), [2](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/modules/net-lb-app-int)) and I think we can benefit here from the same approach.